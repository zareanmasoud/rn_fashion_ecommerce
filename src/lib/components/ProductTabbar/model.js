// @flow

export type Specs = {
  action: Function,
  src: string,
  text: string,
};
