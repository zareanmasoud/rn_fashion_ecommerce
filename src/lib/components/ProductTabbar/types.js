// @flow

export type Button = {
  handlePress: Function,
  src: string,
  text: string,
};
