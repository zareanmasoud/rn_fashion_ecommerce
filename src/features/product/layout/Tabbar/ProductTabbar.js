import React, {useContext} from 'react';
import {withNavigation} from 'react-navigation';
import ProductTabbarComponent from 'lib/components/ProductTabbar';
import specs from './specs';
import CartContext from '../../../../context/CartContext';

const ProductTabbar = () => {
  const {addToCart} = useContext(CartContext);

  return (
    <ProductTabbarComponent
      buttons={[specs.buttons.addWishlist, specs.buttons.fitting]}
      addToCart={specs.buttons.addToCart(addToCart)}
    />
  );
};

export default withNavigation(ProductTabbar);
