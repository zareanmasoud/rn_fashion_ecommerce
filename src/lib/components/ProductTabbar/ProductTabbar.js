import React from 'react';
import {View} from 'react-native';
import TabbarButton from 'lib/components/TabbarButton';
import AddToCartButton from 'lib/components/AddToCartButton';
import styles from './styles';
import type {Specs} from './model';

type Props = {
  specs: Specs[],
};

const ProductTabbar = ({specs}: Props) => {
  const specsArray = Object.values(specs);
  const addToCart = specsArray.pop();
  const TabbarButtons = specsArray.map(specsItem => (
    <TabbarButton {...specsItem} key={specsItem.id} />
  ));

  return (
    <View style={styles.container}>
      {TabbarButtons}
      <AddToCartButton specs={addToCart} />
    </View>
  );
};

export default ProductTabbar;
