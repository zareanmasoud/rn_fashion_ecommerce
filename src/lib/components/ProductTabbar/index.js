import ProductTabbar from './ProductTabbar';

export {ProductTabbar as default};
