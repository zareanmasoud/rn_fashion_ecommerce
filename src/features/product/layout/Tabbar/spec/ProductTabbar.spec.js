import React from 'react';
import {shallow} from 'enzyme';
import ProductTabbar from '../ProductTabbar';

describe('ProductTabbar Feature', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<ProductTabbar />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
