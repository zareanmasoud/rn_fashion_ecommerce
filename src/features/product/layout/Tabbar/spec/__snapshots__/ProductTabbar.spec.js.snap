// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTabbar Feature Rendering should match to snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
